cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -216,3 +217,8 @@ func (c Client) WaitForClusterAvailable(delay time.Duration, retries int, retryI
 	}
 	return false, fmt.Errorf("timeout waiting for etcd cluster to be available")
 }
+
+// CheckConfigurationIsHA returns true if the given MasterConfiguration etcd block appears to be an HA configuration.
+func CheckConfigurationIsHA(cfg *kubeadmapi.Etcd) bool {
```

Comment:
nit: we call `MasterConfiguration` generally `cfg`. `etcd` or `etcdCfg` or `etcdObj` or `e` would probably be better here.